### PR TITLE
v4: Dependencies tool

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Tool to provide Python dependencies for enabled addons on v4 server

Currently implemented:
- validate addon toml that it includes expected keys
- merge two tomls with resolving of versions for `dependencies`, `dev-dependencies`, `thirdparty`
- prepare full toml based on root pyproject.toml and all downloaded addons
- prepare and zip venv based on resolved toml
- zip file contains additional metadata (python version used, platform, hash of poetry.lock)
- preparation to upload zip to v4 server
- 
Contains battery of unit tests that should highlight business logic.

Directory of interest should be `pype/common/openpype_common/distribution/dependencies`. 
Currently this PR contains also code for client delivery tool (for client to download client code for active addons). It is possible that this PR should be targeted to separate repo altogether.

Implemented only for .ps1 for now. (Small updates to `create_env.ps1` - to allow create .venv at specific location. Needs to be translated to .sh if no issues will be raised.)

To be implemented:
- ~proper name of venv~
- download thirparty and figure out how to distribute them (`venv` vs `vendor`) - in future PR
- ~upload zip file to web~ - not real endpoint on server yet
- how to actually call this - currently prepared main endopoint in `dependencies.py.main()` (expects v4 server url)

### CODE REVIEW

**Comment by iLLiCiTiT** [comment]
This file should be removed: `distribution/dependencies/__init__.py`.

File: common/openpype_common/distribution/dependencies/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
+import os.path
```
Comment (iLLiCiTiT) [inline common/openpype_common/distribution/dependencies/dependencies.py:1]:
```suggestion
import os
```

**Comment by iLLiCiTiT** [comment]
Content of `common/openpype_common/distribution/dependencies` probably should be in new repo. It's useless on client.

**Comment by kalisp** [comment]
> Content of `common/openpype_common/distribution/dependencies` probably should be in new repo. It's useless on client.

Need to decide how this will be distributed/deployed. It expects existing `pyproject.toml` and `poetry.lock` for 'base' (or build) venv. (At least because of `requests`)

And I wanted to share `RemoteFileHandler.zip` method. (But thats not a biggie, that function could be yanked out easily.)

**Comment by iLLiCiTiT** [comment]
> Need to decide how this will be distributed/deployed. It expects existing pyproject.toml and poetry.lock for 'base' (or build) venv. (At least because of requests)

It'a tool to create environments and it's ok that it will need it's own environment to be able do that.

> And I wanted to share RemoteFileHandler.zip method. (But thats not a biggie, that function could be yanked out easily.)

Yeah, I think you can copy that 6 lines of code :)

**Comment by kalisp** [comment]
Code moved to separate repo: https://github.com/pypeclub/openpypev4-dependencies-tool

**Pull request closed without merging**